src/blockchain/ain/jellyfish/raw-tx-util.ts

```diff
@@ -58,6 +59,35 @@ export class RawTxUtil {
     return [prevouts, rawScriptHex];
   }
 
+  static parseUnspentUntilAmount(unspent: AddressUnspent[], amount: BigNumber): [Prevout[], BigNumber, string] {
+    let rawScriptHex = '';
+    const prevouts = unspent.map((item): Prevout => {
+      rawScriptHex = item.script.hex;
+      return {
+        txid: item.vout.txid,
+        vout: item.vout.n,
+        value: new BigNumber(item.vout.value),
+        script: {
+          // TODO(fuxingloh): needs to refactor once jellyfish refactor this.
+          stack: toOPCodes(SmartBuffer.fromBuffer(Buffer.from(item.script.hex, 'hex'))),
+        },
+        tokenId: item.vout.tokenId ?? 0x00,
+      };
+    });
+    let total = new BigNumber(0);
+    const neededPrevouts: Prevout[] = [];
+    prevouts.forEach((p) => {
+      if (total.gte(amount)) return;
+      neededPrevouts.push(p);
+      total = total.plus(p.value);
+    });
```

Comment:
Could be combined in a single for-each ... I don't know if it would be more readable 🤔

Reply:
we could move that map into a single function, as this is currently done twice in exactly the same manner.